.github/releaserc.json

```diff
@@ -0,0 +1,31 @@
+{
+  "branches": ["master"],
+  "plugins": [
+    "@semantic-release/commit-analyzer",
+    "@semantic-release/release-notes-generator",
+    [
+      "@semantic-release/exec",
+      {
+        "prepareCmd": "npm ci"
+      }
+    ],
+    [
+      "@semantic-release/npm",
+      {
+        "npmPublish": true
+      }
+    ],
+    [
+      "@semantic-release/changelog",
+      {
+        "changelogTitle": "# Changelog\nAll notable changes to the project will be documented in this file."
+      }
+    ],
+    [
+      "@semantic-release/git",
```

Comment:
- added `assets` in https://github.com/stacks-network/jsontokens-js/compare/b8edc9fc7b75c45d575e2229cb5239368faa69cc..38a02b7e7722f4d9f7f18d1a581d6118f4c9da73

---

⚠️ careful, github actions can be a bit confusing here. i believe, the filters in `on` (e.g. `push`) are OR conditions, which means [for this example](https://github.com/hirosystems/gh-actions-example/blob/main/.github/workflows/ci.yml#L33-L35) any push to `main/develop` will ignore the `*-ignore` conditions and run nonetheless.

refs:
- [docs.github](https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#example-including-branches-and-tags)
- [github.community comment 1](https://github.community/t/using-on-push-tags-ignore-and-paths-ignore-together/16931/9)
- [github.community comment 2](https://github.community/t/github-action-for-master-branch-and-ignore-all-tags-and-first-level-of-files/172557)
- [example stacks.js run (from v tagged commit)](https://github.com/hirosystems/stacks.js/runs/6511545186?check_suite_focus=true)

Reply:
Aha, thanks for the break-down! 🚀 In my head `tags-ignore` would stop a job from running on a commit that matches the pattern (similar to `paths-ignore`), but these are separate events . `on push branches main` will ignore `tags` and `tags-ignore` completely, since it's not a branch push-event (even if the tag was pushed first, and .git _would_ know about it). 🙃 That makes sense!!! Thanks 🙏

Not super necessary here, as we have the `if:` condition checking for HEAD commit message, but better to add the `paths-ignore`, yes 👍

---

_documenting for my future self:_

> If you define both branches/branches-ignore and [paths](https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#onpushpull_requestpull_request_targetpathspaths-ignore), the workflow will only run when both filters are satisfied. -- [docs#](https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#onpushbranchestagsbranches-ignoretags-ignore), [sample-repo](https://github.com/janniks/test-actions/actions)

only paths+branches is an AND condition. while tags vs branches are separate events.

Reply:
Added
```
    paths-ignore:
      - "**.json"
      - "**.md"
```